firebase-database/src/main/java/com/google/firebase/database/FirebaseDatabase.java

```diff
@@ -202,7 +192,9 @@ public DatabaseReference getReferenceFromUrl(@NonNull String url) {
           "Can't pass null for argument 'url' in " + "FirebaseDatabase.getReferenceFromUrl()");
     }
 
-    ParsedUrl parsedUrl = Utilities.parseUrl(url, getEmulatorServiceSettings(this.app));
+    ParsedUrl parsedUrl =
+        Utilities.parseUrl(
+            url, getApp().get(FirebaseDatabaseComponent.class).getEmulatorSettings());
```

Comment:
@vkryachko what's the proper way to do this?  Lint says I shouldn't do this outside of `getInstance` but I can't store a reference to the component in `getInstance` since the instance comes from the component ...

Reply:
In this case `FirebaseDatabaseComponent` is the owner of `FirebaseDatabase` instances and responsible for creating them, so I would just recommend passing this component to the `FirebaseDatabase`'s constructor here:
https://github.com/firebase/firebase-android-sdk/blob/cc79cc638975677eadb5d742a5423312a82debf4/firebase-database/src/main/java/com/google/firebase/database/FirebaseDatabaseComponent.java#L66

Reply:
Ok so it's safe to pass the component down?  Thanks!